### PR TITLE
Make `parse_policy_or_template_to_est` error for CST that aren't valid AST

### PR DESCRIPTION
**Pull request by john-h-kastner-aws**
The EST conversion does not apply the checks that we apply CST to EST conversion, and we don't want to duplicate all of them. For example, this function succeeded for expression using extension function with an incorrect call style like `"1".ip()` (cedar-policy/cedar-spec#341).

Effects `policy_text_to_json` in the WASM bindings.

## Checklist for requesting a review

The change in this PR is (choose one, and delete the other options):

- [x] A change (breaking or otherwise) that only impacts unreleased or experimental code.

I confirm that this PR (choose one, and delete the other options):

- [x] Updates the "Unreleased" section of the CHANGELOG with a description of my change (required for major/minor version bumps).

I confirm that [`cedar-spec`](https://github.com/cedar-policy/cedar-spec) (choose one, and delete the other options):

- [x] Does not require updates because my change does not impact the Cedar formal model

### CODE REVIEW

File: cedar-wasm/CHANGELOG.md [hunk 1]
```diff
@@ -16,3 +16,8 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 - Exposed types through `tsify` for `ValidateCall` and the schema. (#692)
 - Exposed cedar-wasm functionality for formatter and schema, context, and entity parsing: `wasm_format_policies`, `check_parse_schema`, `check_parse_context`, `check_parse_entities`. (#718)
 - Exposed cedar-wasm functionality for template parsing: `check_parse_template`.
+
+### Changed
+
+- Update `policyTextToJson` so that it reliably errors for all policies that
```
Comment (shaobo-he-aws) [inline cedar-wasm/CHANGELOG.md:22]: Are these quoted names functions?

**Review by shaobo-he-aws — APPROVED** [review]
LGTM

File: cedar-wasm/CHANGELOG.md (hunk 1 above)
Comment (john-h-kastner-aws) [reply cedar-wasm/CHANGELOG.md:22]: They're the JS binding function names from annotations like `#[wasm_bindgen(js_name = "checkParseTemplate")]`

Comment (shaobo-he-aws) [reply cedar-wasm/CHANGELOG.md:22]: Got it. Thanks.

**Review by cdisselkoen — APPROVED** [review]

**Review by cdisselkoen — APPROVED** [review]

**Pull request merged**